src/super_gradients/training/dataloaders/dataloaders.py

```diff
@@ -81,14 +82,24 @@ def get_data_loader(config_name: str, dataset_cls: object, train: bool, dataset_
     return dataloader
 
 
-def _process_dataset_params(cfg, dataset_params, train):
-    default_dataset_params = cfg.train_dataset_params if train else cfg.val_dataset_params
-    default_dataset_params = hydra.utils.instantiate(default_dataset_params)
-    for key, val in default_dataset_params.items():
-        if key not in dataset_params.keys() or dataset_params[key] is None:
-            dataset_params[key] = val
+def _process_dataset_params(cfg, dataset_params, train: bool):
+    # Due to some internal details of Hydra, we need to merge the dataset_params with the default dataset params while retaining the
+    # top-level config structure.
+    # We can't do:
+    # >>> default_dataset_params = cfg.train_dataset_params if train else cfg.val_dataset_params
+    # because it will break interpolation down the road.
+    if train:
+        cfg.train_dataset_params = OmegaConf.merge(cfg.train_dataset_params, dataset_params)
+    else:
+        cfg.val_dataset_params = OmegaConf.merge(cfg.val_dataset_params, dataset_params)
+
+    # default_dataset_params = cfg.train_dataset_params if train else cfg.val_dataset_params
```

Comment:
Left on purpose?

Reply:
Yes, I imagine having (for some time) old config merging code can be helpful if we ran into some unexpected problems with `OmegaConf.merge`. For this purpose I left this code block commented to save some time for one debugging it.